iOS/ViewControllers/Courses/LTIExerciseHintViewController.swift

```diff
@@ -82,22 +105,19 @@ class LTIExerciseHintViewController: UIViewController {
             self.exerciseTypeLabel.text = nil
         }
 
-        // Set points label
-        let format = NSLocalizedString("course-item.max-points", comment: "maximum points for course item")
-        let number = NSNumber(value: self.courseItem.maxPoints)
-        self.pointsLabel.text = Self.pointsFormatter.string(from: number).flatMap { String.localizedStringWithFormat(format, $0) }
+        self.exerciseTypeView.isHidden = false
 
-        // Set time effort label
-        let roundedTimeEffort = ceil(TimeInterval(self.courseItem.timeEffort) / 60) * 60 // round up to full minutes
-        self.timeEffortLabel.text = Self.timeEffortFormatter.string(from: roundedTimeEffort)
-        self.timeEffortView.isHidden = self.courseItem.timeEffort == 0
+        guard let ltiExercise = self.courseItem?.content as? LTIExercise else { return }
 
-        // Set deadline label
-        self.deadlineLabel.text = self.courseItem.deadline.map(Self.dateFormatter.string(from:))
-        self.deadlineDateView.isHidden = self.courseItem.deadline == nil
+        func hideLoadingScreen() {
+            self.loadingScreen.alpha = 0.0
+        }
 
-        // Set instructions label
-        self.instructionsView.setMarkdownWithImages(from: ltiExercise.instructions)
+        UIView.animate(withDuration: 0.5, delay: 0.0, options: .curveLinear, animations: hideLoadingScreen) { _ in
+            self.loadingScreen.isHidden = true
+            self.launchExerciseView.isHidden = false
+            self.instructionsView.setMarkdownWithImages(from: ltiExercise.instructions)
+        }
```

Comment:
Instead of adding an animation when switching to the model data, I would rather like to see an animation added when the loading screen is present (appear, duration: 0.25, delay: 0.25). By doing so and using a slight delay, we will avoid showing the loading spinner when there is a quick response by the backend.